crates/project-model/src/project_json.rs

```diff
@@ -122,6 +122,27 @@ impl ProjectJson {
                         None => None,
                     };
 
+                    crate_data.cfg_groups.sort_unstable();
+                    crate_data.cfg_groups.dedup();
+                    let cfg = crate_data
+                        .cfg_groups
+                        .iter()
+                        .flat_map(|cfg_extend| {
+                            let cfg_group = data.cfg_groups.get(cfg_extend);
+                            match cfg_group {
+                                Some(cfg_group) => cfg_group.0.iter().cloned(),
+                                None => {
+                                    tracing::error!(
+                                        "Unknown cfg group `{cfg_extend}` in crate `{}`",
+                                        crate_data.display_name.as_deref().unwrap_or("<unknown>"),
+                                    );
+                                    [].iter().cloned()
```

Comment:
More of a large meta point, but I think it's be a good idea to us to consider using https://github.com/rust-lang/annotate-snippets-rs for rust-analyzer-specific errors.

Reply:
That seems tailored to CLI output, doesn't sound too useful for us